spec.html

```diff
@@ -23844,17 +24404,41 @@ <h1>HostImportModuleDynamically ( _referencingScriptOrModule_, _specifier_, _pro
+        <h1>FinishDynamicImport ( _referencingScriptOrModule_, _specifier_, _promiseCapability_, _innerPromise_ )</h1>
+        <p>The abstract operation FinishDynamicImport takes arguments _referencingScriptOrModule_, _specifier_, _promiseCapability_ (a PromiseCapability Record), and _innerPromise_. FinishDynamicImport completes the process of a dynamic import originally started by an <emu-xref href="#sec-import-calls">`import()`</emu-xref> call, resolving or rejecting the promise returned by that call as appropriate according to _innerPromise_'s resolution. It is performed by host environments as part of HostImportModuleDynamically. It performs the following steps when called:</p>
         <emu-alg>
-          1. If _completion_ is an abrupt completion, perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; _completion_.[[Value]] &raquo;).
-          1. Else,
-            1. Assert: _completion_ is a normal completion and _completion_.[[Value]] is *undefined*.
-            1. Let _moduleRecord_ be ! HostResolveImportedModule(_referencingScriptOrModule_, _specifier_).
-            1. Assert: Evaluate has already been invoked on _moduleRecord_ and successfully completed.
-            1. Let _namespace_ be GetModuleNamespace(_moduleRecord_).
-            1. If _namespace_ is an abrupt completion, perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; _namespace_.[[Value]] &raquo;).
-            1. Else, perform ! Call(_promiseCapability_.[[Resolve]], *undefined*, &laquo; _namespace_.[[Value]] &raquo;).
+          1. Let _stepsFulfilled_ be the steps of a CallDynamicImportResolved function as specified below.
+          1. Let _onFulfilled_ be CreateBuiltinFunction(_stepsFulfilled_, 0, *""*, &laquo; [[ReferencingScriptOrMoule]], [[Specifier]], [[PromiseCapability]] &raquo;).
+          1. Set _onFulfilled_.[[ReferencingScriptOrModule]] to _referencingScriptOrModule_.
+          1. Set _onFulfilled_.[[Specifier]] to _specifier_.
+          1. Set _onFulfilled_.[[PromiseCapability]] to _promiseCapability_.
+          1. Let _stepsRejected_ be the steps of a CallDynamicImportRejected function as specified below.
+          1. Let _onRejected_ be CreateBuiltinFunction(_stepsRejected_, 0, *""*, &laquo; [[PromiseCapability]] &raquo;).
+          1. Set _onRejected_.[[PromiseCapability]] to _promiseCapability_.
+          1. Perform ! PerformPromiseThen(_innerPromise_.[[Promise]], _onFulfilled_, _onRejected_).
+        </emu-alg>
+
+        <p>A CallDynamicImportResolved function is an anonymous built-in function with the internal
+        slots [[ReferencingScriptOrModule]], [[Specifier]], and [[PromiseCapability]], and it takes
+        one argument, _result_. When a CallDynamicImportResolved function is called it performs the following steps:</p>
```

Comment:
This `<p>` element should be one line.

Also some suggested wording changes:
```suggestion
        <p>A CallDynamicImportResolved function is an anonymous built-in function with the internal slots [[ReferencingScriptOrModule]], [[Specifier]], and [[PromiseCapability]]. It takes one argument, _result_. When called, it performs the following steps:</p>
```